examples/perceive_and_act/custom_agents.py

```diff
@@ -0,0 +1,79 @@
+import logging
+import os
+import threading
+import time
+from wiseagents import WiseAgent, WiseAgentMessage, WiseAgentTransport
+from wiseagents.yaml import WiseAgentsLoader
+
+class PerceivingAgent(WiseAgent):
```

Comment:
Although this agent doesn't make use of things like process_request, process_response, etc., it should still implement the 4 abstract methods with something like just `pass` in the body of the implementation as described here:

https://github.com/wise-agents/wise-agents/blob/main/docs/custom_agents.md